cura/Machines/Models/MaterialManagementModel.py

```diff
@@ -264,3 +270,52 @@ def removeFavorite(self, material_base_file: str) -> None:
             self.favoritesChanged.emit(material_base_file)
         except ValueError:  # Material was not in the favorites list.
             Logger.log("w", "Material {material_base_file} was already not a favorite material.".format(material_base_file = material_base_file))
+
+    def _onOutputDevicesChanged(self) -> None:
+        """
+        When the list of output devices changes, we may want to update the
+        preferred export path.
+        """
+        cura_application = cura.CuraApplication.CuraApplication.getInstance()
+        device_manager = cura_application.getOutputDeviceManager()
+        devices = device_manager.getOutputDevices()
+        for device in devices:
+            if device.__class__.__name__ == "RemovableDriveOutputDevice":
```

Comment:
Oof. I'm really not a fan of this :( Unfortunately all the other solutions that I can come up with are worse.

Reply:
An alternative would've been to check the device's short description (`Save to removable drive`, but translated) or its icon (`save_sd`). We can't use the device name because it's arbitrary, named after the actual drive name. The stream is just a file stream. The rest of the real properties are not unique.

The short description is a bit finicky because it's translated. It would work but not be nice code either. The icon is easy but not very descriptive. I opted for this code because it's clear what it does.

We can't use `isinstance` or something like that without importing classes from a plug-in.

One alternative would be to define a new field, like `OutputDeviceTargetType` and have that be a string by which we can identify it. But yeah, the class name is kind of already that. It's not a big improvement.

Reply:
As discussed online, we're leaving it like this.